### ISSUE TITLE
Inconsistent Node.js behavior for handling `set-cookie` headers

### ISSUE THREAD

**Issue by pi0** [nitrojs/nitro#1462]
Discovered while adding tests in https://github.com/unjs/nitro/pull/1452

- Node.js < 18, merges all `set-cookie` headers (seems also inconsistent between `18.12` and `18.16` too https://github.com/unjs/nitro/issues/1581)
- Node.js >= 18 keeps them separated but does not also auto split

**Comment by Hebilicious** [comment]
Given that Node 18 reach EOL in a 3 months` should we consider moving everything to node 20 ? 🤔

**Comment by pi0** [comment]
Good point. Still we support both. Maybe we can reduce amount of parsing workaround in favor of Node 20 :)

**Comment by Hebilicious** [comment]
> Good point. Still we support both. Maybe we can reduce amount of parsing workaround in favor of Node 20 :)

Vite 5 is moving to drop support for node < 18 as well, so I think it makes sense that we align and support 18 and 20. 
I would like to move to 20+ though 😓

**Comment by pi0** [comment]
Surely there is no blocker for Nuxt ecosystem at least. We preannounced rolling major versions for v3 (https://nuxt.com/blog/v3#the-browser-and-nodejs-support)

Only for nitro, i think i will aim for next major version to drop node 16,18.

**Comment by jeanpierrecarvalho** [comment]
@pi0 @Hebilicious Given that Node.js versions 17 and 19 have reached their end of life (EOL), I propose adjusting the Node.js version requirement to only accept versions >=18.16, while explicitly excluding version 19, and also including support for version 20. This adjustment would align with the EOL status of previous versions and ensure a more robust and forward-looking compatibility. What are your thoughts on this?

**Comment by pi0** [comment]
@jeanpierrecarvalho  As explained above, it is not that easy decision to make to immediately drop older EOL versions of Node.js we have a vast ecosystem and users.

Also even if we drop older Node.js versions at some point, runtime behavior of recent Node.js is not consistent and does not automatically split cookies.

**Comment by Hebilicious** [comment]
@pi0 Do we have access to the telemetry stats regarding the node versions that nitro/nuxt / nitro preset users are using ? Having some insight would be helpful in making these type of decisions.

**Comment by pi0x** [comment]
Multiple set-cookie headers stay separate in v3: on a built node-server app, both handler-set and proxied cookies come back as distinct headers. The Node <18 merging path is gone since v3 requires Node ^20.19 || >=22.12, and test/tests.ts covers this across presets.

_Triaged with AI assistance._
